pkg/resources/task_acceptance_test.go

```diff
@@ -306,6 +304,303 @@ func TestAcc_Task_Updates(t *testing.T) {
+						HasCommentString(comment).
+						HasFinalizeString("").
+						HasAfterIdsInOrder().
+						HasWhenString(condition).
+						HasSqlStatementString(newStatement),
+					resourceshowoutputassert.TaskShowOutput(t, completeConfigModel.ResourceReference()).
+						HasCreatedOnNotEmpty().
+						HasName(id.Name()).
+						HasIdNotEmpty().
+						HasDatabaseName(id.DatabaseName()).
+						HasSchemaName(id.SchemaName()).
+						HasOwner(currentRole.Name()).
+						HasWarehouse(warehouse.ID()).
+						HasComment(comment).
+						HasScheduleMinutes(5).
+						HasPredecessors().
+						HasState(sdk.TaskStateStarted).
+						HasDefinition(newStatement).
+						HasCondition(condition).
+						HasAllowOverlappingExecution(true).
+						HasErrorIntegration(errorNotificationIntegration.ID()).
+						HasLastCommittedOnNotEmpty().
+						HasLastSuspendedOn("").
+						HasOwnerRoleType("ROLE").
+						HasConfig(taskConfig).
+						HasBudget("").
+						HasTaskRelations(sdk.TaskRelations{}),
+				),
+			},
+			// Unset
+			{
+				Config: config.FromModel(t, basicConfigModel),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction(basicConfigModel.ResourceReference(), plancheck.ResourceActionUpdate),
+					},
+				},
+				Check: assert.AssertThat(t,
+					resourceassert.TaskResource(t, basicConfigModel.ResourceReference()).
+						HasFullyQualifiedNameString(id.FullyQualifiedName()).
+						HasDatabaseString(id.DatabaseName()).
+						HasSchemaString(id.SchemaName()).
+						HasNameString(id.Name()).
+						HasStartedString(r.BooleanFalse).
+						HasWarehouseString("").
+						HasNoScheduleSet().
+						HasConfigString("").
+						HasAllowOverlappingExecutionString(r.BooleanDefault).
+						HasErrorIntegrationString("").
+						HasCommentString("").
+						HasFinalizeString("").
+						HasAfterIdsInOrder().
+						HasWhenString("").
+						HasSqlStatementString(statement),
+					resourceshowoutputassert.TaskShowOutput(t, basicConfigModel.ResourceReference()).
+						HasCreatedOnNotEmpty().
+						HasName(id.Name()).
+						HasIdNotEmpty().
+						HasDatabaseName(id.DatabaseName()).
+						HasSchemaName(id.SchemaName()).
+						HasOwner(currentRole.Name()).
+						HasComment("").
+						HasWarehouse(sdk.NewAccountObjectIdentifier("")).
+						HasNoSchedule().
+						HasPredecessors().
+						HasState(sdk.TaskStateSuspended).
+						HasDefinition(statement).
+						HasCondition("").
+						HasAllowOverlappingExecution(false).
+						HasErrorIntegration(sdk.NewAccountObjectIdentifier("")).
+						HasLastCommittedOnNotEmpty().
+						HasLastSuspendedOnNotEmpty().
+						HasOwnerRoleType("ROLE").
+						HasConfig("").
+						HasBudget("").
+						HasTaskRelations(sdk.TaskRelations{}),
+				),
+			},
+		},
+	})
+}
+
+/*
+DAG structure (the test proves child3 won't have any issues with updates in the following scenario):
+
+		 child1
+		/		\
+	 root 		 child3
+		\		/
+		 child2
```

Comment:
Nit: root should be on top

Reply:
I would leave it this way, formatter doesn't like these kinds of comments and when changing the structure of current graph it formats strangely.